docs/topics/signals.rst

```diff
@@ -54,6 +54,23 @@ Some signals support returning :class:`~twisted.internet.defer.Deferred`
 objects from their handlers, see the :ref:`topics-signals-ref` below to know
 which ones.
 
+.. note::
+
+    If a signal handler returns a :class:`~twisted.internet.defer.Deferred`,
+    the spider waits for that :class:`~twisted.internet.defer.Deferred` to fire.
+    The signal handler can also return a :class:`~twisted.internet.defer.Deferred`
+    obtained from some async operation which will fire as the operation is finished.
```

Comment:
I think this could be merged into the previous paragraph, and the reference to the list of signals be moved after the example.

For example:

> Some signals support returning :class:`~twisted.internet.defer.Deferred`
objects from their handlers, allowing you to run asynchronous code that does not block Scrapy. If a signal handler returns a :class:`~twisted.internet.defer.Deferred`, Scrapy waits for that :class:`~twisted.internet.defer.Deferred` to fire.
>
> [example]
>
> See the :ref:`topics-signals-ref` below to know which signals support Deferreds.

I’m not sure the example is very helpful:
- `getDeferredFromSomewhere()` makes me wonder where I can get a Deferred from, which is something that is not explained.
- I find the comment in the `return d` line confusing, and the other comment, above, meaningless (stating the obvious while users may not even understand how to define a callback).

An example based on a real use case, but simplified, may be better. See what @wRAR did at https://github.com/scrapy/scrapy/pull/4329/files#diff-8ff82039e3841d26573a59dfa988882eR53, for example. Maybe a similar example can be used handling one of the item signals. Although a different use case may be better. Maybe something as simple as using `treq` to send a request “out of Scrapy” to a fictional API?

In general, though, I thing these changes are OK; just mentioning “asynchronous” improves the current documentation.

In the long term, though, I think we may want to consider writing a topic of its own about running asynchronous code in Scrapy, which covers some use cases, useful libraries, different syntax (Scrapy 2.0.0 coroutines), etc. But that may be better for a different PR.

Reply:
Thanks for the review. I will update the docs.

Reply:
@Gallaecio I have updated the docs using a simpler example which uses `treq` to send scrapped item to the server.